fix(integration): createTask({ projectId }) returns task with projectId=null

## Context

`createTask({ name, projectId })` followed by `getTask(taskId)` returns a task whose `projectId` is `null`. The task is created but its parent-project association is lost (or never written, or never read back). This is the canonical failure in the long-standing 100% failure rate of the Integration Tests workflow on every release tag since v1.0.0 — confirmed reproducible **locally** against my OmniFocus, not just on the self-hosted runner.

Most likely root cause for the ~5–7 cascading failures (`moveTask`, `reorderTask` ×3, `duplicateTask` ×3, `moveProject`) — they all start by creating a task with a `projectId` and then failing the moment they look it up.

**Latest failed run:** [run 25129921989](https://github.com/torsday/omnifocus-mcp/actions/runs/25129921989) (15 of 48 failed). **Local repro:**

```bash
OMNIFOCUS_INTEGRATION=1 pnpm vitest run \
  tests/integration/transportRouter.contract.test.ts \
  -t "createTask with projectId places the task"
# → AssertionError: expected null to be 'dgqXDtMxX7u'
```

Tracks: cluster of integration test failures repro'd in run 25129921989.

## Acceptance Criteria

- [ ] `createTask({ name, projectId })` followed by `getTask(taskId)` returns a task whose `projectId` matches the input
- [ ] `tests/integration/transportRouter.contract.test.ts > tasks — CRUD > createTask with projectId places the task in that project` passes locally and on the `macos-omnifocus` runner
- [ ] The cascading failures that share this root cause also flip green: `moveTask into a project updates projectId`, `reorderTask — at:start moves task to start of container`, `reorderTask — at:end moves task to end of container`, `reorderTask — { at, in } to a different project reparents`, `duplicateTask — clones a task alongside the source by default`, `duplicateTask — recursive clones full subtree depth-first`, `duplicateTask — destination.projectId reparents the clone`, `moveProject to a different folder updates folderId`, `tagId filter returns tasks carrying that tag`, `updateProject applies a patch`, `listProjects filters by status`. (If any survive the fix, file follow-up tickets per the bug.)
- [ ] Root cause documented in commit message (which JXA script was broken and how)

## Technical Notes

- Wrapper at [src/adapter/jxa/JxaTransport.ts:223](src/adapter/jxa/JxaTransport.ts#L223) passes `projectId` to the JXA script correctly — the bug is **not** in TypeScript glue.
- Suspect [scripts/jxa/task_create.js](scripts/jxa/task_create.js) (parent-project assignment via `assignedContainer` or similar) or [scripts/jxa/task_get.js](scripts/jxa/task_get.js) (how `containingProject.id()` is read back). Compare against [scripts/jxa/task_update.js](scripts/jxa/task_update.js)'s reparent path which is presumably correct.
- Routing table at [src/adapter/router.ts:96](src/adapter/router.ts#L96) shows `createTask: "jxa"` — JXA handles create, OmniJS handles move. Mixed-transport interaction is one possible failure surface.
- Closed issues #279–#282 dealt with similar JXA `.make()` regressions during initial integration-suite bring-up — worth re-reading the diffs that fixed them.
- The fix should land **before** the next release; the integration suite is supposed to be the final pre-publish guard but has been silently red for the entire v1.x line.

## Dependencies

- Tracks: cluster of integration test failures repro'd in [run 25129921989](https://github.com/torsday/omnifocus-mcp/actions/runs/25129921989)
- Probably parent-of (i.e. fixing this likely closes): the cascading task/project/tag failures in the same suite
- Related issue (orthogonal failure cluster, different root cause): error-mapping regression where JXA NotFound returns ScriptError

--- a/src/scripts/jxa/forecast_get.js
+++ b/src/scripts/jxa/forecast_get.js
@@ -48,10 +48,19 @@ function run(argv) {
   }
 
   function buildTask(task) {
+    // See task_get.js — `cp.class()` throws on real projects in OF 4.x (#673).
     let projectId = null;
     try {
       const cp = task.containingProject();
-      if (cp && cp.class() !== "document") projectId = cp.id();
+      if (cp) {
+        let isDocument = false;
+        try {
+          isDocument = cp.class() === "document";
+        } catch (_classErr) {
+          /* OF 4.x: real projects throw here */
+        }
+        if (!isDocument) projectId = cp.id();
+      }
     } catch (_e) {}
 
     let parentId = null;

--- a/src/scripts/jxa/perspective_evaluate.js
+++ b/src/scripts/jxa/perspective_evaluate.js
@@ -36,10 +36,19 @@ function run(argv) {
     }
 
     function buildTask(task) {
+      // See task_get.js — `cp.class()` throws on real projects in OF 4.x (#673).
       let projectId = null;
       try {
         const cp = task.containingProject();
-        if (cp && cp.class() !== "document") projectId = cp.id();
+        if (cp) {
+          let isDocument = false;
+          try {
+            isDocument = cp.class() === "document";
+          } catch (_classErr) {
+            /* OF 4.x: real projects throw here */
+          }
+          if (!isDocument) projectId = cp.id();
+        }
       } catch (_e) {}
 
       let parentId = null;

--- a/src/scripts/jxa/task_create.js
+++ b/src/scripts/jxa/task_create.js
@@ -34,10 +34,21 @@ function run(argv) {
   }
 
   function buildTask(task) {
+    // See task_get.js — `cp.class()` throws on real projects in OF 4.x
+    // (#673). Treat a throw as "real project", a successful return as
+    // "document — skip".
     let projectId = null;
     try {
       const cp = task.containingProject();
-      if (cp && cp.class() !== "document") projectId = cp.id();
+      if (cp) {
+        let isDocument = false;
+        try {
+          isDocument = cp.class() === "document";
+        } catch (_classErr) {
+          /* OF 4.x: real projects throw here */
+        }
+        if (!isDocument) projectId = cp.id();
+      }
     } catch (_e) {}
 
     let parentId = null;

--- a/src/scripts/jxa/task_duplicate.js
+++ b/src/scripts/jxa/task_duplicate.js
@@ -37,9 +37,21 @@ function run(argv) {
     if (pt) {
       destContainer = pt;
     } else {
+      // See task_get.js — `cp.class()` throws on real projects in OF 4.x
+      // (#673), and the throw used to escape this if-condition unhandled,
+      // exiting the whole script with `JXA script failed (exit 1)`.
       const cp = source.containingProject();
-      if (cp && cp.class() !== "document") destContainer = cp;
-      else destContainer = doc;
+      if (cp) {
+        let isDocument = false;
+        try {
+          isDocument = cp.class() === "document";
+        } catch (_classErr) {
+          /* OF 4.x: real projects throw here */
+        }
+        destContainer = isDocument ? doc : cp;
+      } else {
+        destContainer = doc;
+      }
     }
   }
 

--- a/src/scripts/jxa/task_get.js
+++ b/src/scripts/jxa/task_get.js
@@ -29,10 +29,26 @@ function run(argv) {
   }
 
   function buildTask(task) {
+    // OmniFocus 4.x note (#673): on a real Project specifier returned by
+    // `task.containingProject()`, calling `.class()` throws "Can't convert
+    // types". Only the document responds to `.class()` successfully. So:
+    // treat a `.class()` throw as "is a real project", a successful return
+    // as "is something else (document) — skip". The previous one-liner
+    // (`cp.class() !== "document"`) let the throw escape the if-condition
+    // and the outer try/catch swallowed the whole block, leaving projectId
+    // null even when the task was correctly assigned to a project.
     let projectId = null;
     try {
       const cp = task.containingProject();
-      if (cp && cp.class() !== "document") projectId = cp.id();
+      if (cp) {
+        let isDocument = false;
+        try {
+          isDocument = cp.class() === "document";
+        } catch (_classErr) {
+          /* OF 4.x: real projects throw here — leave isDocument false */
+        }
+        if (!isDocument) projectId = cp.id();
+      }
     } catch (_e) {}
 
     let parentId = null;

--- a/src/scripts/jxa/task_get_many.js
+++ b/src/scripts/jxa/task_get_many.js
@@ -29,10 +29,19 @@ function run(argv) {
   }
 
   function buildTask(task) {
+    // See task_get.js — `cp.class()` throws on real projects in OF 4.x (#673).
     let projectId = null;
     try {
       const cp = task.containingProject();
-      if (cp && cp.class() !== "document") projectId = cp.id();
+      if (cp) {
+        let isDocument = false;
+        try {
+          isDocument = cp.class() === "document";
+        } catch (_classErr) {
+          /* OF 4.x: real projects throw here */
+        }
+        if (!isDocument) projectId = cp.id();
+      }
     } catch (_e) {}
 
     let parentId = null;

--- a/src/scripts/jxa/task_list.js
+++ b/src/scripts/jxa/task_list.js
@@ -35,10 +35,19 @@ function run(argv) {
   }
 
   function buildTask(task) {
+    // See task_get.js — `cp.class()` throws on real projects in OF 4.x (#673).
     let projectId = null;
     try {
       const cp = task.containingProject();
-      if (cp && cp.class() !== "document") projectId = cp.id();
+      if (cp) {
+        let isDocument = false;
+        try {
+          isDocument = cp.class() === "document";
+        } catch (_classErr) {
+          /* OF 4.x: real projects throw here */
+        }
+        if (!isDocument) projectId = cp.id();
+      }
     } catch (_e) {}
 
     let parentId = null;

--- a/src/scripts/jxa/task_search.js
+++ b/src/scripts/jxa/task_search.js
@@ -46,10 +46,19 @@ function run(argv) {
   }
 
   function buildTask(task) {
+    // See task_get.js — `cp.class()` throws on real projects in OF 4.x (#673).
     let projectId = null;
     try {
       const cp = task.containingProject();
-      if (cp && cp.class() !== "document") projectId = cp.id();
+      if (cp) {
+        let isDocument = false;
+        try {
+          isDocument = cp.class() === "document";
+        } catch (_classErr) {
+          /* OF 4.x: real projects throw here */
+        }
+        if (!isDocument) projectId = cp.id();
+      }
     } catch (_e) {}
 
     let parentId = null;

--- a/src/scripts/jxa/task_update.js
+++ b/src/scripts/jxa/task_update.js
@@ -35,10 +35,19 @@ function run(argv) {
   }
 
   function buildTask(task) {
+    // See task_get.js — `cp.class()` throws on real projects in OF 4.x (#673).
     let projectId = null;
     try {
       const cp = task.containingProject();
-      if (cp && cp.class() !== "document") projectId = cp.id();
+      if (cp) {
+        let isDocument = false;
+        try {
+          isDocument = cp.class() === "document";
+        } catch (_classErr) {
+          /* OF 4.x: real projects throw here */
+        }
+        if (!isDocument) projectId = cp.id();
+      }
     } catch (_e) {}
 
     let parentId = null;

